From b59c48ce61eeca56742a4164f42139997cf84363 Mon Sep 17 00:00:00 2001
From: Adrian Ratiu <adrian.ratiu@collabora.com>
Date: Tue, 21 May 2024 11:52:05 +0300
Subject: [PATCH 3/3] meson.build: install lib and headers

This ensures lib and headers are properly installed in
meson build, without having to hardcode for eg the builddir.

Upstream-Status: In progress [https://github.com/libtypec/libtypec/pull/1]

Signed-off-by: Adrian Ratiu <adrian.ratiu@collabora.com>
---
 meson.build       |  7 +++++++
 utils/meson.build | 25 +++++++++++++++++++++----
 2 files changed, 28 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 7a7c28a..903c506 100644
--- a/meson.build
+++ b/meson.build
@@ -21,4 +21,11 @@ library('typec',
 	version : meson.project_version(),
 	soversion : '5',
 	dependencies: libudev_dep,
+	install: true,
+)
+
+install_headers(
+	'libtypec.h',
+	meson.current_build_dir() + '/libtypec_config.h',
+	install_dir: 'include'
 )
diff --git a/utils/meson.build b/utils/meson.build
index 9bd2245..7796d0a 100644
--- a/utils/meson.build
+++ b/utils/meson.build
@@ -12,13 +12,30 @@ conf_data.set('libtypec_utils_VERSION_PATCH', split[2])
 
 configure_file(input : 'libtypec_utils_config.h.in', output : 'libtypec_utils_config.h', configuration : conf_data)
 
+# Include current build dir for the above generated file
+inc_dir = include_directories('.')
+
 cc = meson.get_compiler('c')
 
 dep = declare_dependency(
-  dependencies : cc.find_library('typec', dirs : [meson.current_source_dir()]),
-  include_directories : include_directories('../../builddir/'),
+	dependencies : cc.find_library('typec')
 )
+
 udev_dep = meson.get_compiler('c').find_library('udev')
 
-executable('lstypec', 'lstypec.c', 'names.c' ,dependencies : [dep,udev_dep])
-executable('typecstatus', 'typecstatus.c', 'names.c',dependencies : [dep,udev_dep])
+executable(
+	'lstypec',
+	'lstypec.c', 'names.c',
+	dependencies: [dep, udev_dep],
+	include_directories: inc_dir,
+	install: true,
+	install_dir: get_option('bindir')
+)
+executable(
+	'typecstatus',
+	'typecstatus.c', 'names.c',
+	dependencies: [dep, udev_dep],
+	include_directories: inc_dir,
+	install: true,
+	install_dir: get_option('bindir')
+)
-- 
2.44.1

